lib/template.js

```diff
@@ -0,0 +1,202 @@
+        templateLicense = "// This file is auto generated and will be overwritten \n" +
+        "// @licstart  The following is the entire license notice for the \n" +
+        "//  JavaScript code in this page.\n" +
+        "// \n" +
+        "// Copyright 2011-2013, E14N https://e14n.com/\n" +
+        "// \n" +
+        "// Licensed under the Apache License, Version 2.0 (the \"License\");\n" +
+        "// you may not use this file except in compliance with the License.\n" +
+        "// You may obtain a copy of the License at\n" +
+        "// \n" +
+        "//     http://www.apache.org/licenses/LICENSE-2.0\n" +
+        "// \n" +
+        "// Unless required by applicable law or agreed to in writing, software\n" +
+        "// distributed under the License is distributed on an \"AS IS\" BASIS,\n" +
+        "// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n" +
+        "// See the License for the specific language governing permissions and\n" +
+        "// limitations under the License.\n" +
+        "// \n" +
+        "// @licend  The above is the entire license notice\n" +
+        "// for the JavaScript code in this page.\n\n" +
+        "// XXX: this needs to be broken up into 3-4 smaller modules \n";
+
+
+    // Get al files from views directory but only take all .jade files
+    _.each(viewsFiles, function(file) {
+        var _template = getTemplate(file);
+        if (_template) {
+            // Add template compile function with hash version
+            templates[_template.name] = {
+                hash: _template.hash,
+                string: _template.string
+            };
+            templatesHashes.push("\"" + _template.name + "\": " + "\"" + _template.hash + "\"");
+        }
+    });
+
+    // Return hash by template
+    var templatesCode = "if (!window.Pump) {window.Pump = {};}" +
+        "(function(Pump) {\"use strict\";" +
+        " Pump._templates = { " + templatesHashes.join(",") + "};" +
+        "})(window.Pump);";
+
+    // Compress
+    if (!opts.debug) {
+        templatesCode = uglify.minify(templatesCode, {
+            fromString: true
+        }).code;
+        outputName += ".min";
+    }
+    // Add license after because uglify remove comments
+    templatesCode = templateLicense + templatesCode;
+    outputName += ".js";
+
+    // Get templates
+    function getTemplate (file) {
+        try {
+            var templatePath = path.join(viewsPath, file),
+                stat = fs.lstatSync(templatePath);
+
+
+            if (stat.isFile() && !blackListRegexp.test(file) && /\.jade$/.test(file)) {
+                var templateName = path.basename(file, ".jade"),
+                    templateContent = fs.readFileSync(templatePath, {encoding: "utf8"}),
+                    // Compile jade template for client side
+                    templateFn = jade.compileClient(templateContent, {
+                        externalRuntime: false,
```

Comment:
What is this option? I don't see it in [the docs](https://pugjs.org/api/reference.html); maybe you meant `inlineRuntimeFunctions`?

Reply:
In jade 1.11.x both options`externalRuntime` or `inlineRuntimeFunctions` not exist, see: https://github.com/pugjs/pug/tree/master/packages/pug-runtime https://github.com/pugjs/pug/commit/1eeb9e6ea37a638601cc5559c37bf86ea4749b13